### PR TITLE
Fix http2cat sample

### PR DESCRIPTION
**Pull request by amcasey**
As of #41558, four settings are expected, not three.

~Bonus: Actually show the http2cat logging~

### CODE REVIEW

**Comment by amcasey** [comment]
Specifically, https://github.com/dotnet/aspnetcore/pull/41558/files#diff-aa7215e5bc089bf5d35990bd3640949ef03eed4004a29b07e102e489c18fd11fR127

File: src/Servers/Kestrel/samples/http2cat/Program.cs [hunk 1]
```diff
@@ -28,7 +30,7 @@ public static async Task Main(string[] args)
 
     internal static async Task RunTestCase(Http2Utilities h2Connection)
     {
-        await h2Connection.InitializeConnectionAsync();
+        await h2Connection.InitializeConnectionAsync(expectedSettingsCount: 4);
```
Comment (BrennanConroy) [inline src/Servers/Kestrel/samples/http2cat/Program.cs:33]: Change https://github.com/amcasey/aspnetcore/blob/bc9255b509c4b51d9bbbea7b15ddafe482784699/src/Shared/Http2cat/Http2Utilities.cs#L153 instead?

Comment (BrennanConroy) [reply src/Servers/Kestrel/samples/http2cat/Program.cs:33]: IIS tests use Http2Utilities so we can't change the default.

File: src/Servers/Kestrel/samples/http2cat/Program.cs [hunk 2]
```diff
@@ -18,6 +18,8 @@ public static async Task Main(string[] args)
         using var host = new HostBuilder()
             .ConfigureLogging(loggingBuilder =>
             {
+                loggingBuilder.AddFilter((category, level) =>
+                    level >= LogLevel.Information || category.StartsWith("Microsoft.AspNetCore.Http2Cat", StringComparison.Ordinal));
```
Comment (BrennanConroy) [inline src/Servers/Kestrel/samples/http2cat/Program.cs:22]: I thought the default was info?

Comment (amcasey) [reply src/Servers/Kestrel/samples/http2cat/Program.cs:22]: Empirically, not including that seems to drop everything that doesn't match "Http2Cat".

Comment (BrennanConroy) [reply src/Servers/Kestrel/samples/http2cat/Program.cs:22]: I see Http2Cat logs locally when running http2cat sample.

> info: Microsoft.AspNetCore.Http2Cat.Http2CatHostedService[0]
      Connected to '[::1]:5001'.
info: Microsoft.AspNetCore.Http2Cat.Http2CatHostedService[0]
      Starting TLS handshake.
info: Microsoft.AspNetCore.Http2Cat.Http2CatHostedService[0]
      TLS handshake completed successfully.
info: Microsoft.AspNetCore.Http2Cat.Http2CatHostedService[0]
      Initialized http2 connection. Starting stream 1.

Comment (amcasey) [reply src/Servers/Kestrel/samples/http2cat/Program.cs]: It was probably just missing because of the exception I was seeing...

**Review by BrennanConroy — APPROVED** [review]

**Pull request merged**